### PR TITLE
Add a fluent api to work with fg/bg colors and modifiers

### PR DESCRIPTION
**Pull request by KennedyTedesco**
Partially closes: https://github.com/php-tui/php-tui/issues/109

With this feature, instead of:

```php
Span::styled('[q]', Style::default()->fg(AnsiColor::Red))
```

We just:

```php
Span::fromString('[q]')->red()
```

Next step (a new PR): I'll implement a parser, something like:

```php
Line::parse('<fg=green options=bold,italic>Hello</> World')
```

### CODE REVIEW

File: src/Model/Style/InteractsWithBgColor.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+declare(strict_types=1);
+
+namespace PhpTui\Tui\Model\Style;
+
+use PhpTui\Tui\Model\AnsiColor;
+use PhpTui\Tui\Model\Color;
+use PhpTui\Tui\Model\Style;
+
+trait InteractsWithBgColor
```
Comment (dantleech) [inline src/Model/Style/InteractsWithBgColor.php:11]: is there a reason to have 3 traits? either something can have a style or it cannot? I would call it `StyleableTrait`.

Comment (KennedyTedesco) [reply src/Model/Style/InteractsWithBgColor.php:11]: The idea was to have more focused, smaller traits. A `StyleableTrait` with all methods looks good to me either. No objections. =D

**Comment by dantleech** [comment]
:+1: looks good - not sure we need three different traits though?

File: src/Model/Style/InteractsWithBgColor.php (hunk 1 above)
Comment (dantleech) [reply src/Model/Style/InteractsWithBgColor.php:11]: let's have `StyleableTrait` and if we need to break it down later we can. note that there is also just one test called `StylebaleTest` so it makes sense from that POV :slightly_smiling_face:

Comment (KennedyTedesco) [reply src/Model/Style/InteractsWithBgColor.php]: I agree!

**Comment by dantleech** [comment]
:tada: nice!

**Pull request merged**